### PR TITLE
[KYUUBI #1905] log4j2-defaults.properties is not working since log4j 2 is always intialized by default

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

As per [SPARK-37746](https://github.com/apache/spark/pull/35026), Kyuubi also has the same problem with initializing Log4j2.

[log4j2](https://github.com/apache/logging-log4j2/blob/rel/2.17.1/log4j-core/src/main/java/org/apache/logging/log4j/core/config/AbstractConfiguration.java#L717) provides a default configuration so there is always an appender (ConsoleAppender) with ERROR level

This pr changes the behavior of checking the initialization of log4j2.

```
LogManager.getRootLogger
  .asInstanceOf[org.apache.logging.log4j.core.Logger]
  .getAppenders
  .asScala
  .foreach {
    case (name, appender) =>
      println(s"default appender: $name, class: ${appender.getClass.getName}")
  }
```

```
default appender: DefaultConsole-2, class: org.apache.logging.log4j.core.appender.ConsoleAppender
```

After this pr, the root level is `INFO`.

截屏2022-02-14 下午5 32 26

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by cfmcgrady** [comment]
cc @pan3793 @yaooqinn @turboFei @link3280

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala [hunk 1]
```diff
@@ -179,4 +179,20 @@ object Logging {
     val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
     "org.apache.logging.slf4j.Log4jLoggerFactory".equals(binderClass)
   }
+
+  /**
+   * Return true if log4j2 is initialized by default configuration which has one
+   * appender with error level. See `org.apache.logging.log4j.core.config.DefaultConfiguration`.
+   */
+  private[kyuubi] def islog4j2DefaultConfigured(): Boolean = {
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala:187]: private?

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala [hunk 2]
```diff
@@ -92,16 +93,15 @@ trait Logging {
...
       // scalastyle:off println
-      if (!log4j2Initialized) {
+      if (Logging.islog4j2DefaultConfigured()) {
         Logging.useDefault = true
         val defaultLogProps = "log4j2-defaults.properties"
         Option(Thread.currentThread().getContextClassLoader.getResource(defaultLogProps)) match {
           case Some(url) =>
             val context = LogManager.getContext(false).asInstanceOf[LoggerContext]
             context.setConfigLocation(url.toURI)
+            System.err.println(s"Using Kyuubi's default log4j profile: $url")
```
Comment (yaooqinn) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala:104]: looks unnecessary

**Comment by yaooqinn** [comment]
I have tested it locally, LGTM overall.

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala (hunk 2 above)
Comment (cfmcgrady) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala:104]: Only for debug, it's ok for me to revert.

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-common/src/main/resources/log4j2-defaults.properties [hunk 3]
```diff
@@ -29,7 +29,7 @@ appender.console.layout.pattern = %d{HH:mm:ss.SSS} %p %c: %m%n
 appender.console.filter.1.type = Filters
 
 appender.console.filter.1.a.type = ThresholdFilter
-appender.console.filter.1.a.level = fatal
+appender.console.filter.1.a.level = INFO
```
Comment (link3280) [inline kyuubi-common/src/main/resources/log4j2-defaults.properties:32]: Minor: should we keep in lower case to align with others?

**Review by link3280 — COMMENTED** [review]
@cfmcgrady Thanks for the PR. LGTM in general, with a minor concern of mine.

File: kyuubi-common/src/main/resources/log4j2-defaults.properties (hunk 3 above)
Comment (jiwq) [reply kyuubi-common/src/main/resources/log4j2-defaults.properties:32]: Agree to @link3280 . If need change, should better modify global.

File: kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala [hunk 4]
```diff
@@ -179,4 +178,20 @@ object Logging {
     val binderClass = StaticLoggerBinder.getSingleton.getLoggerFactoryClassStr
     "org.apache.logging.slf4j.Log4jLoggerFactory".equals(binderClass)
   }
+
+  /**
+   * Return true if log4j2 is initialized by default configuration which has one
+   * appender with error level. See `org.apache.logging.log4j.core.config.DefaultConfiguration`.
+   */
+  private def islog4j2DefaultConfigured(): Boolean = {
```
Comment (jiwq) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/Logging.scala:186]:
```suggestion
  private def isLog4j2DefaultConfigured(): Boolean = {
```

**Review by link3280 — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks @cfmcgrady for the fix and @jiwq @link3280 for the review, merged to master

**Pull request closed without merging**